docs/plugins/development/background-jobs.md

```diff
@@ -46,25 +57,49 @@ As described above, jobs can be scheduled for immediate execution or at any late
 
 #### Example
 
+```python title="models.py"
+from django.db import models
+from core.choices import JobIntervalChoices
+from netbox.models import NetBoxModel
+from .jobs import MyTestJob
+
+class MyModel(NetBoxModel):
+    foo = models.CharField()
+
+    def save(self, *args, **kwargs):
+        MyTestJob.enqueue_once(instance=self, interval=JobIntervalChoices.INTERVAL_HOURLY)
+        return super().save(*args, **kwargs)
+
+    def sync(self):
+        MyTestJob.enqueue(instance=self)
+```
+
+
+### System Jobs
+
+Some plugins may implement background jobs that are decoupled from any object and the request-response cycle. Typical use cases would be housekeeping tasks or synchronization jobs. These can be created using *system jobs*. The `JobRunner` class has everything included to provide this type of job as well. Just add the appropriate metadata to let NetBox schedule all background jobs automatically.
+
+!!! info
+    All system jobs are automatically scheduled just before the `./manage.py rqworker` command is started and the job queue is processed. The schedules are also checked at each restart of this process.
+
+#### Example
+
 ```python title="jobs.py"
+from core.choices import JobIntervalChoices
 from netbox.jobs import JobRunner
-
+from .models import MyModel
 
 class MyHousekeepingJob(JobRunner):
```

Comment:
I think an import to MyHousekeepingJob still needs to be in __init even if there is only a system_job to run (no other background jobs?  Also, I noticed separate issue the Example in the docs line ` MyHousekeepingJob.setup(interval=60)` I think is incorrect as there is no setup() method in the JobRunner and it was throwing an error for me when I had copied the code?

That said I wasn't actually getting the job firing on schedule when I added it as a test to a plugin I had installed, not sure if I had it mis-configured. In the plugins jobs.py:

```
from core.choices import JobIntervalChoices
from netbox.jobs import JobRunner

class MyHousekeepingJob(JobRunner):
    class Meta:
        name = "My Housekeeping Job"
        system_interval = JobIntervalChoices.INTERVAL_MINUTELY

    def run(self, *args, **kwargs):
        print("in run...")

system_jobs = (
    MyHousekeepingJob,
)
```
In the __init:
```
from netbox.plugins import PluginConfig

class HealthCheckConfig(PluginConfig):
    name = 'netbox_healthcheck_plugin'
    verbose_name = 'NetBox HealthCheck Plugin'
    description = 'NetBox plugin for HealthCheck.'
    version = 'version'
    base_url = 'netbox_healthcheck_plugin'
    django_apps = [
        'health_check',
        'health_check.db',
        'health_check.contrib.migrations',
        'health_check.contrib.redis',
    ]
    min_version = "v4.0-beta1"
    events_pipeline = ["netbox_healthcheck_plugin.events.process_healthcheck_queue"]

    def ready(self):
        from .jobs import MyHousekeepingJob
        # MyHousekeepingJob.setup()

config = HealthCheckConfig

```
I was expecting the run( to get hit every minute, but I wasn't ever seeing it run. Could be operator error.

Reply:
The documentation for `setup()` was fixed in 39e5c0d24c700d251ecf5a49c2bcd068d09472b6. This was left over from removing system background jobs from the original FR.

`HealthCheckConfig.ready()` is not needed. The job should be enqueued immediately when the `rqworker` process starts. Did you restart the worker after update?

Reply:
As registration now uses a decorator, I think this may be outdated.